### PR TITLE
[Doc FR] — All files — French Translation Done

### PR DESCRIPTION
**Pull request by MachinisteWeb**
We are the [Vue-FR](https://github.com/vuejs/vuejs.org#french) team. We provide our translation on Vue Router part (with cross consistency from others Vue documentations).

Translated files:
- All!

Process:
- We will track all futur modifications from `en` directory because in our `working` branch we use the `en` directory to work. All futur `en` modifications at an already translated position will warns us with a conflict.
- We reverse all work done from `:working/en` to `:dev/fr` and we will provide some pull request each time a (set of) page(s) will be translated/updated.

### CODE REVIEW

**Comment by MachinisteWeb** [comment]
Why CircleCl failed? We have only add `.md` files to `fr`

**Comment by MachinisteWeb** [comment]
Hi @posva. 

I just update some changes from original. This PR is ready for merge.

What can I do to allows this PR to be merged?

**Comment by MachinisteWeb** [comment]
Hi @posva !

The current documentation is half in french, half in english on the https://router.vuejs.org/. This PR bring ALL french translation with official french terms consistency with others Vue documentation.

It's ready to merge !

**Comment by posva** [comment]
I'll give it a look as soon as I can.
edit: will proly be next week

**Comment by MachinisteWeb** [comment]
Hi @posva 

I have translated the new `navigation.md` part in french.

All files are ready for merge as soon as possible :)

**Comment by MachinisteWeb** [comment]
I have also change two minor consistency things in EN doc.

An unecessary return line and a `2.2` that's become `2.2+`

File: docs/fr/advanced/navigation-guards.md [hunk 1]
```diff
@@ -70,65 +70,65 @@ const router = new VueRouter({
...
+L'interception `beforeRouteEnter` **n'**a **PAS** accès à `this`, car l'interception est appelée avant que la navigation soit confirmée, et le nouveau composant entrant n'a même pas encore été crée.
 
-However, you can access the instance by passing a callback to `next`. The callback will be called when the navigation is confirmed, and the component instance will be passed to the callback as the argument:
+Cependant, vous pouvez accéder à l'instance en passant dans la fonction de rappel `next`. Cette fonction de rappel va être appelée quand la navigation sera confirmée, et l'instance du composant sera passée à la fonction de rappel en tant qu'argument :
 
 ``` js
 beforeRouteEnter (to, from, next) {
   next(vm => {
-    // access to component instance via `vm`
+    // accèr à l'instance du composant via `vm`
```
Comment (posva) [inline docs/fr/advanced/navigation-guards.md]: accès

File: docs/fr/essentials/navigation.md [hunk 2]
```diff
@@ -30,8 +30,22 @@ router.push({ name: 'user', params: { userId: 123 }})
...
+router.push({ path: `/user/${userId}` }) // -> /user/123
+// Ceci ne vas PAS fonctionner
+router.push({ path: '/user', params: { userId }}) // -> /user
+```
+
+Les mêmes règles s'appliquent pour la propriété `to` du composant `router-link`.
+
 Dans la version 2.2.0+, vous pouvez optionnellement fournir les fonctions de rappel `onComplete` et `onAbort` à `router.push` ou `router.replace` en tant que deuxième et troisième arguments. Ces fonctions de rappel seront appelées quand la navigation sera respectivement ; complétée avec succès (après la résolution de tous les hooks asynchrones), ou arrêtée (navigation vers la même route ou vers une route différente avant que la navigation courante ne soit achevée).
 
+**Note:** If the destination is the same as the current route and only params are changing (eg: going from one profile to another `/users/1` -> `/users/2`), you will have to use [beforeRouteUpdate](./dynamic-matching.html#reacting-to-params-changes) to react to changes (eg: fetching the user information).
```
Comment (posva) [inline docs/fr/essentials/navigation.md]: Elle s'est échapé, cette note 😄

**Review by posva — CHANGES_REQUESTED** [review]

**Comment by MachinisteWeb** [comment]
Merci @posva ! C'est fait.

File: docs/fr/essentials/navigation.md (hunk 2 above)
Comment (MachinisteWeb) [reply docs/fr/essentials/navigation.md]: Je l'ai rattrapée !

**Review by posva — APPROVED** [review]
Merci!

**Comment by MachinisteWeb** [comment]
Super ! 🎆

**Comment by MachinisteWeb** [comment]
Hi @posva !

What is the frequency of update for https://router.vuejs.org/fr/?

I'm currently waiting to propose French translation to French public via social networks etc. because currently there is still the previous content (half English / half French).

I can do something?

**Comment by posva** [comment]
I deployed it!

**Pull request merged**